### PR TITLE
KFSPTS 12390 Stale Check Job

### PR DESCRIPTION
**Pull request by ajd299**

### CODE REVIEW

File: src/test/java/com/rsmart/kuali/kfs/cr/batch/fixture/StaleCheckRowFixture.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    CHECK_123_DISB_STAL_29252_VALID("20180809", CrTestConstants.MELLON_BANK_CODE, CRConstants.STALE, "12345678", "292.52", true),
+    CHECK_123_BLANK_STAL_19223_INVALID("20180609", KFSConstants.EMPTY_STRING, CRConstants.STALE, "82345678", "192.23", false),
+    CHECK_199_DISB_STAL_212319_VALID("20180712", CrTestConstants.MELLON_BANK_CODE, CRConstants.STALE, "19999999", "2,123.19", true),
+    CHECK_399_DISB_VOID_951("20180712", CrTestConstants.MELLON_BANK_CODE, CRConstants.VOIDED, "39999999", "9.51", false),
+    CHECK_111_DISB_CLRD_123("20180715", CrTestConstants.MELLON_BANK_CODE, CRConstants.CLEARED, "11111111", "1.23", false),
+    CHECK_211_DISB_STOP_123("20180715", CrTestConstants.MELLON_BANK_CODE, CRConstants.STOP, "21111111", "1.23", false),
+    CHECK_311_DISB_STAL_123("20180715", CrTestConstants.MELLON_BANK_CODE, CRConstants.STALE, "31111111", "1.23", false),
+    CHECK_411_DISB_EXCP_123("20180715", CrTestConstants.MELLON_BANK_CODE, CRConstants.EXCP, "41111111", "1.23", false),
+    CHECK_511_DISB_CDIS_321("20180715", CrTestConstants.MELLON_BANK_CODE, CRConstants.CANCELLED, "51111111", "3.21", false);
```
Comment (cah292) [inline src/test/java/com/rsmart/kuali/kfs/cr/batch/fixture/StaleCheckRowFixture.java]: There are several fixtures in this particular set that are expecting an issue date of "20180715", even though the associated test file is using "20180712" instead. Is such a discrepancy intentional for this test case?

Comment (ajd299) [reply src/test/java/com/rsmart/kuali/kfs/cr/batch/fixture/StaleCheckRowFixture.java]: Changed to 20180712 for consistency.

File: src/main/java/edu/cornell/kfs/pdp/service/impl/CuPendingTransactionServiceImpl.java [hunk 2]
```diff
@@ -267,6 +269,25 @@ else if (paymentGroup.getDisbursementType().getCode().equals(PdpConstants.Disbur
...
+        String checkYear = KFSConstants.getDashSubAccountNumber();
+        try {
+            CheckReconciliation checkReconciliation = checkReconciliationDao.findByCheckNumber(paymentGroup.getDisbursementNbr().toString(), paymentGroup.getBankCode());
+            Calendar calendar = Calendar.getInstance();
+            if (ObjectUtils.isNotNull(checkReconciliation) && ObjectUtils.isNotNull(checkReconciliation.getCheckDate())) {
+                calendar.setTime(checkReconciliation.getCheckDate());
+                checkYear = Integer.toString(calendar.get(Calendar.YEAR));
+            }
+            else {
+                LOG.error("getCheckYear", "Failed getting year check was issued, defaulting clearing subaccount to dashes.");
```
[the remark this answers is not in the document]
Comment (ajd299) [reply src/main/java/edu/cornell/kfs/pdp/service/impl/CuPendingTransactionServiceImpl.java]: Changed to "getCheckYear: Failed...".

**Comment by ajd299** [comment]
I have made some changes to address your comments. Ready for re-review.

File: src/test/java/com/rsmart/kuali/kfs/cr/batch/service/impl/StaleCheckExtractServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+                StaleCheckRowFixture expectedStaleCheckRow = expectedRows.get(i);
+                StaleCheckBatchRow actualStaleCheckRow = actualRows.get(i).staleCheckBatchRow;
+
+                verifyParsedRowMatchesExpectedResult(expectedStaleCheckRow, actualStaleCheckRow);
+            }
+        }
+    }
+
+    private void verifyParsedRowMatchesExpectedResult(StaleCheckRowFixture expectedRow, StaleCheckBatchRow actualRow) {
+        if (expectedRow.valid) {
```
Comment (cah292) [inline src/test/java/com/rsmart/kuali/kfs/cr/batch/service/impl/StaleCheckExtractServiceImplTest.java]: Is an extra assertion needed in this method (or in the caller method), to check that the batch row succeeded or failed as expected?

**Comment by ajd299** [comment]
Thanks the latest suggestion flushed out a possible issue with commas. As a result I switched from built-in KualiDecimal constructor to a java.text parser that can handle commas. Please review again

**Pull request merged**